### ISSUE TITLE
AM/PM parsing bug in some German texts

### ISSUE THREAD

**Issue by infostreams** [facebook/duckling#196]
Hi!

First of all, thanks for this amazing piece of software 🙏

Second, I noticed a parsing error while testing this software on some German text. Specifically, the following text:

"Wir treffen uns am 17 Uhr am KiLa."

is misinterpreted. When fed to Duckling, I receive the following response:

```json
{  
   "body":"am 17 Uhr am ",
   "start":16,
   "value":{  
      "values":[  
         {  
            "value":"2018-05-18T05:00:00.000+02:00",
            "grain":"hour",
            "type":"value"
         },
         {  
            "value":"2018-05-19T05:00:00.000+02:00",
            "grain":"hour",
            "type":"value"
         },
         {  
            "value":"2018-05-20T05:00:00.000+02:00",
            "grain":"hour",
            "type":"value"
         }
      ],
      "value":"2018-05-18T05:00:00.000+02:00",
      "grain":"hour",
      "type":"value"
   },
   "end":29,
   "dim":"time",
   "latent":false
}
```

This is unambiguously wrong. What is happening is that the recognized section includes "am" after the time, and Duckling assumes that it means "Ante Meridiem" in this context and therefore converts the "17 Uhr" (which is clearly PM, not AM) into "05:00". Instead, "am" in this context means "in the". It has nothing to do with the time and should not be matched.

My suggestion would be to either not apply AM/PM corrections to blatantly unambiguous times such as "17:00", or to not parse AM/PM at all in German texts. These AM/PM markers are not used in German - at all. They are culturally Anglo-Saxon, and the only times you see them are in English texts. Germans don't use them. In German, 'am' means 'an dem' in 99+% of the cases, not "ante meridiem".

**Comment by chinmay87** [comment]
@infostreams Thanks for bringing this up!
The rule `<time-of-day> am|pm` can be removed from German as it is not relevant

**Comment by infostreams** [comment]
I just submitted a pull request to remove this rule. With that adjustment the problematic original sentence 

    "Wir treffen uns am 17 Uhr am KiLa."

now returns the correct response:

```json
{  
   "body":"am 17 Uhr",
   "start":75,
   "value":{  
      "values":[  
         {  
            "value":"2018-05-22T17:00:00.000+02:00",
            "grain":"hour",
            "type":"value"
         },
         {  
            "value":"2018-05-23T17:00:00.000+02:00",
            "grain":"hour",
            "type":"value"
         },
         {  
            "value":"2018-05-24T17:00:00.000+02:00",
            "grain":"hour",
            "type":"value"
         }
      ],
      "value":"2018-05-22T17:00:00.000+02:00",
      "grain":"hour",
      "type":"value"
   },
   "end":84,
   "dim":"time",
   "latent":false
}
```

**Comment by infostreams** [comment]
('start' and 'end' keys are off because there's other text before and after the problematic sentence)

**Comment by xhavokx** [comment]
Closed by ddbb6fdff8dfe0e1d6b20946c57bac38c79601c7
Thanks for providing the PR @infostreams.

**Comment by infostreams** [comment]
You're welcome. Deleting code is easy 😊 Thanks for writing the tests!
